### ISSUE TITLE
Support default font for bold/italic style?

### ISSUE THREAD

**Issue by akhy** [chrisjenx/Calligraphy#66]
I don't know if this feature has been implemented or not. I just recently applying `OpenSans-Regular.ttf` as global default, then all bold TextView turned to normal (non-bold style). I wish there is a built-in and clean way to provide default bold font file (e.g. `OpenSans-Bold.ttf`) when used by `TextView`, perhaps by checking it in `TextView#getTypeface().isBold()`.

I think this feature will come in handy because it will eliminate the need to apply custom style one by one to all bold/italic texts.

Thanks

**Comment by vyndor** [comment]
Any chance to see it soon? Title and buttons for material appcompat AlertDialog aren't bold anymore when I inject context with Calligraphy.

**Comment by chrisjenx** [comment]
Thanks because they will be inflated using your default font.

Soon? unlikely, this is something I don't want to hack together.

You can override default styles which would let you set the title correctly
etc.

**Comment by yayaa** [comment]
Any improvements on this thread?

I had same issue, and trying to find which default styles i need to override was kinda painful for AlertDialog... So here how you can do it: 


    <style name="AppTheme" parent="Theme.AppCompat.Light.NoActionBar">
       ...
        <item name="alertDialogTheme">@style/AlertDialogStyle</item>
    </style>
    
    <style name="AlertDialogStyle" parent="Theme.AppCompat.Light.Dialog.Alert">
        <item name="colorAccent">@color/colorAccent</item>
        <item name="android:windowTitleStyle">@style/DialogWindowTitleStyle</item>
    </style>
    
    <style name="DialogWindowTitleStyle" parent="@android:style/TextAppearance.DialogWindowTitle">
        <item name="android:textAppearance">@style/CustomFontStyle.Bold</item>
    </style>

     <style name="CustomFontStyle" parent="TextAppearance.AppCompat" />

     <style name="CustomFontStyle.Bold" parent="TextAppearance.AppCompat" >
         <item name="fontPath">fonts/bold_font.ttf</item>
     </style>

**Comment by ramona-cristea** [comment]
Was added any solution to support bold style?

**Comment by chrisjenx** [comment]
I'm currently discussing this with some other engineers, no ETA yet. Sorry.

**Comment by nirslsk** [comment]
For some reason using the <b> and <i> tags works with Calligraphy fonts in TextViews when used like this:

```
someTextView.setText(Html.fromHtml("<b>something in bold</b> <i>something in italics</i>"));
```

Even seems to work with some very unusual fonts I tried (Blzee, Hemi-Head, Noteworthy, World of Water), not sure why...

**Comment by chrisjenx** [comment]
Yeah this is called text paint bold and fake italic. It works, but not true
bold and italic, if it works for you go for it

**Comment by A-w-K** [comment]
Is anyone working on a solution for this?

**Comment by chrisjenx** [comment]
@A-w-K not only more, this lib is deprecated in favour of https://github.com/InflationX and even that will get depreciated when the new support lib is released

**Comment by omidraha** [comment]
> @akhyrul Yes I have been researching how to do this. I expect this will be a 2.x feature.
> 
> For example you would define:
> 
> ```
> fonts/
>   OpenSans-Regular.ttf
>   OpenSans-Bold.ttf
>   OpenSans-Italic.ttf
> ```
> 
> Which would scope to something like:
> 
> ```
> <TextView
>   fontPath="fonts/OpenSans-*" 
>   textStyle="bold"/>
> ```
> 
> Which would inherently lookup `OpenSans-Bold` if it can't find it then apply fake bold etc.
> 
> This would also be applicable to strings `<string>Hello <b>World</b></string>` which would try to find the correct bold version instead of applying fake bold.
> 
> This is what you're thinking right?

Is this implemented in [Calligraphy3](https://github.com/InflationX/Calligraphy) ?

**Comment by chrisjenx** [comment]
It's not, but we'll take prs for calligraphy 3 as that supports Android x
where as this doesn't.

Lemme know if you open a PR over their so I can review.

Thanks!
